### PR TITLE
[ASP.NET Core] Remove `http.host` and add `net.host.name` and `net.host.port`

### PR DESCRIPTION
**Pull request by vishweshbankwar**
Towards #3373 

## Changes

Spec: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#http-server-semantic-conventions

For significant contributions please make sure you have completed the following items:

* [x] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

**Review by Kielek — APPROVED** [review]

File: src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs [hunk 1]
```diff
@@ -198,14 +198,10 @@ public void OnStartActivity(Activity activity, object payload)
                 activity.DisplayName = path;
 
                 // see the spec https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md
-
-                if (request.Host.Port is null or 80 or 443)
-                {
-                    activity.SetTag(SemanticConventions.AttributeHttpHost, request.Host.Host);
-                }
-                else
+                activity.SetTag(SemanticConventions.AttributeNetHostName, request.Host.Host);
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs:201]: need any check to confirm if Host is set?

**Review by cijothomas — APPROVED** [review]

File: src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs (hunk 1 above)
Comment (vishweshbankwar) [reply src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs:201]: not sure, will have to check.

File: src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+* **Breaking change** `http.host` will no longer be populated. `net.host.name`
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md:5]: please make it clear if this is for traces or metrics.

File: src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs (hunk 1 above)
Comment (vishweshbankwar) [reply src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs]: added a check

File: src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs [hunk 3]
```diff
@@ -198,14 +198,14 @@ public void OnStartActivity(Activity activity, object payload)
...
-
-                if (request.Host.Port is null or 80 or 443)
+                if (request.Host.HasValue)
                 {
-                    activity.SetTag(SemanticConventions.AttributeHttpHost, request.Host.Host);
+                    activity.SetTag(SemanticConventions.AttributeNetHostName, request.Host.Host);
                 }
-                else
+
+                if (request.Host.Port is not null && request.Host.Port != 80 && request.Host.Port != 443)
```
[the remark this answers is not in the document]
Comment (vishweshbankwar) [reply src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs]: Yes - I missed it. Fixed.

**Pull request merged**